src/main/java/com/minecolonies/entity/pathfinding/AbstractPathJob.java

```diff
@@ -900,23 +913,32 @@ else if (block.getMaterial(this.world.getBlockState(pos)).isLiquid())
 
     protected boolean isPassable(BlockPos pos)
     {
-        return isPassable(world.getBlockState(pos).getBlock(), pos);
+        return isPassable(world.getBlockState(pos));
     }
 
     /**
      * Is the block solid and can be stood upon?
      *
-     * @param block Block to check
-     * @param pos   position of block
+     * @param blockState Block to check
      * @return true if the block at that location can be walked on.
      */
-    protected boolean isWalkableSurface(@NotNull Block block, BlockPos pos)
+    protected SurfaceType isWalkableSurface(@NotNull IBlockState blockState)
     {
-        return block.getMaterial(this.world.getBlockState(pos)).isSolid()
-                 && !(block instanceof BlockFence)
-                 && !(block instanceof BlockFenceGate)
-                 && !(block instanceof BlockWall)
-                 && !(block instanceof BlockHutField);
+        Block block = blockState.getBlock();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'block' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)